### DISCUSSION TITLE
Can prisma have types without code gen?

### DISCUSSION THREAD

**Discussion by ricardo-valero** [prisma/prisma#9316] [Q&A] [answered]
I love prisma but I'm not a fan of code generation. 
Would it possible to write prisma and having types without `prisma gen`?
Maybe something similar to [giraphql](https://github.com/hayes/giraphql)?
I don't know much about this or how it works, but it is a far better dev experience than [nexus](https://github.com/graphql-nexus/nexus).

**Comment by nahtnam** [comment, marked as answer]
It is definitely not possible to have Prisma's types without running the generate command.

I'm not 100% sure but I believe even if you do not want the types (using regular node.js), you will still have to run the generate command. Here is a guide to getting started without typescript: https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/relational-databases-node-postgres

In one of the pages it says:

> Whenever you make changes to your Prisma schema in the future, you manually need to invoke prisma generate in order to accommodate the changes in your Prisma Client API.

**Comment by ryands17** [comment]
Prisma is different from Nexus. You can use Prisma with `giraphql`, `TypeGraphQL` or any other GraphQL service. The types that you are talking about are coming from Nexus. Prisma's types are mainly for interacting directly with the database and have nothing to do with GraphQL.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
